lib/code_corps/github/sync/issue/task/changeset.ex

```diff
@@ -23,77 +23,104 @@ defmodule CodeCorps.GitHub.Sync.Issue.Task.Changeset do
     |> Changeset.put_change(:github_repo_id, github_repo_id)
     |> Changeset.put_change(:project_id, project_id)
-    |> Changeset.put_change(:task_list_id, task_list_id)
+    |> assign_task_list(github_issue, project_id)
     |> Changeset.put_change(:user_id, user_id)
     |> Changeset.validate_required([:project_id, :task_list_id, :title, :user_id])
     |> Changeset.assoc_constraint(:github_issue)
     |> Changeset.assoc_constraint(:github_repo)
     |> Changeset.assoc_constraint(:project)
-    |> Changeset.assoc_constraint(:task_list)
     |> Changeset.assoc_constraint(:user)
-    |> Task.order_task()
+    |> maybe_archive()
+    |> Task.handle_archived()
   end
 
   @update_attrs ~w(markdown modified_at status title)a
-  @spec update_changeset(Task.t, map) :: Changeset.t
-  defp update_changeset(%Task{} = task, %{} = issue_attrs) do
+  @spec update_changeset(Task.t, GithubIssue.t, ProjectGithubRepo.t) :: Changeset.t
+  defp update_changeset(
+    %Task{} = task,
+    %GithubIssue{} = github_issue,
+    %ProjectGithubRepo{project_id: project_id}) do
     task
-    |> Changeset.cast(IssueAdapter.to_task(issue_attrs), @update_attrs)
+    |> Changeset.cast(github_issue |> IssueAdapter.to_task, @update_attrs)
     |> MarkdownRendererService.render_markdown_to_html(:markdown, :body)
     |> Changeset.put_change(:modified_from, "github")
     |> TimeValidator.validate_time_not_before(:modified_at)
+    |> assign_task_list(github_issue, project_id)
     |> Changeset.validate_required([:project_id, :title, :user_id])
     |> Changeset.assoc_constraint(:github_repo)
     |> Changeset.assoc_constraint(:project)
     |> Changeset.assoc_constraint(:user)
-    |> Task.order_task()
+    |> maybe_archive()
+    |> Task.handle_archived()
+  end
+
+  @spec assign_task_list(Changeset.t, GithubIssue.t, integer) :: Changeset.t
+  defp assign_task_list(
+    %Changeset{} = changeset, %GithubIssue{} = github_issue, project_id) do
+
+    task_list_type = github_issue |> get_task_list_type()
+    %TaskList{id: task_list_id} = project_id |> get_task_list(task_list_type)
+
+    changeset
+    |> Changeset.put_change(:task_list_id, task_list_id)
+    |> Changeset.assoc_constraint(:task_list)
+  end
+
+  @spec get_task_list_type(GithubIssue.t) :: atom
+  defp get_task_list_type(%GithubIssue{state: "closed"}), do: :done
+  defp get_task_list_type(%GithubIssue{state: "open", github_pull_request_id: pr_id})
+    when not is_nil(pr_id), do: :pull_requests
+  defp get_task_list_type(%GithubIssue{state: "open"}), do: :inbox
+
+  @spec get_task_list(integer, atom) :: TaskList.t
+  defp get_task_list(project_id, :done) do
+    TaskList |> Repo.get_by(project_id: project_id, done: true)
+  end
+  defp get_task_list(project_id, :inbox) do
+    TaskList |> Repo.get_by(project_id: project_id, inbox: true)
+  end
+  defp get_task_list(project_id, :pull_requests) do
+    TaskList |> Repo.get_by(project_id: project_id, pull_requests: true)
+  end
```

Comment:
It could be possible to reverse the `project_id` and `atom` here so that you could chain above:

```elixir
%TaskList{id: task_list_id} =
  github_issue
  |> get_task_list_type()
  |> get_task_list(project_id)
```